src/ui/common/src/components/workflows/artifact/check/history.tsx

```diff
@@ -49,13 +51,20 @@ const CheckHistory: React.FC<CheckHistoryProps> = ({
     schema: checkHistorySchema,
     data: (historyWithLoadingStatus.results?.results ?? []).map(
       (artifactStatusResult) => {
+        let timestamp = new Date(
+          artifactStatusResult.exec_state?.timestamps?.finished_at
+        ).toLocaleString();
+
+        // Checks that are canceled / fail to execute have no exec_state or finished_at time.
+        if (timestamp === 'Invalid Date') {
+          timestamp = 'Unknown';
```

Comment:
Do we have access to the execution status so that we can put that in here instead of "Unknown"? Might be useful to see that a column had errored out or was canceled versus unknown which doesn't have as much information.

Reply:
@eunice-chan The status should actually appear as an icon. This text here is used to show the date in which the check finished execution.

Since there is no finished_at for a check that is canceled or fails to execute, we do not have a date to show here.

Reply:
If you check the render function in this file, you will see that there is a StatusIndicator being rendered for each item in the list:
```
<StatusIndicator
    status={stringToExecutionStatus(entry.status as string)}
    size={'16px'}
    monochrome={false}
/>
```